application/controllers/admin/Database.php

```diff
@@ -277,571 +224,95 @@ private function actionUpdateDefaultValues($iSurveyID)
             }
-
-            $oSurvey->googleanalyticsstyle = $this->filterEmptyFields($oSurvey, 'googleanalyticsstyle');
-            $oSurvey->tokenlength = $this->filterEmptyFields($oSurvey, 'tokenlength');
-            $event = new PluginEvent('beforeSurveySettingsSave');
-            $event->set('modifiedSurvey', $oSurvey);
-            App()->getPluginManager()->dispatchEvent($event);
-            $aAfterApplyAttributes = $oSurvey->attributes;
-
-            if ($oSurvey->save()) {
-                if (!$hasSurveyLanguageSettingError) {
-                    Yii::app()->setFlashMessage(gT("Survey settings were successfully saved."));
-                } else {
-                    Yii::app()->setFlashMessage(gT("Survey settings were saved, but there where errors with some languages."), "warning");
-                }
-            } else {
-                Yii::app()->setFlashMessage(CHtml::errorSummary($oSurvey, CHtml::tag("p", array('class' => 'strong'), gT("Survey could not be updated, please fix the following error:"))), "error");
+            if (!empty($langInput)) {
+                $input[$langCode] = $langInput;
             }
         }
-        $oSurvey->refresh();
 
-        // Url params in json
-        if (Yii::app()->request->getPost('allurlparams', false) !== false) {
-            $aURLParams = json_decode(Yii::app()->request->getPost('allurlparams', ''), true) ?? [];
-            SurveyURLParameter::model()->deleteAllByAttributes(array('sid' => $iSurveyID));
-            foreach ($aURLParams as $aURLParam) {
-                $aURLParam['parameter'] = trim((string) $aURLParam['parameter']);
-                if ($aURLParam['parameter'] == '' || !preg_match('/^[a-zA-Z_][a-zA-Z0-9_]*$/', $aURLParam['parameter']) || $aURLParam['parameter'] == 'sid' || $aURLParam['parameter'] == 'newtest' || $aURLParam['parameter'] == 'token' || $aURLParam['parameter'] == 'lang') {
-                    continue; // this parameter name seems to be invalid - just ignore it
-                }
-                $aURLParam['targetqid']  = $aURLParam['qid'];
-                $aURLParam['targetsqid'] = $aURLParam['sqid'];
-                unset($aURLParam['actionBtn']);
-                unset($aURLParam['title']);
-                unset($aURLParam['id']);
-                unset($aURLParam['qid']);
-                unset($aURLParam['targetQuestionText']);
-                unset($aURLParam['sqid']);
-                if ($aURLParam['targetqid'] == '') {
-                    $aURLParam['targetqid'] = null;
-                }
-                if ($aURLParam['targetsqid'] == '') {
-                    $aURLParam['targetsqid'] = null;
-                }
-                $aURLParam['sid'] = $iSurveyID;
 
-                $param = new SurveyURLParameter();
-                foreach ($aURLParam as $k => $v) {
-                                    $param->$k = $v;
-                }
-                $param->save();
-            }
+        $meta = [];
```

Comment:
What does `meta` mean? :)

Reply:
Meta data. The method returns data about what it did. Specifically the fields that it updated.

Reply:
OK perhaps write a comment about that in the code?